mllib/src/test/scala/org/apache/spark/ml/FunctionsSuite.scala

```diff
@@ -61,5 +61,20 @@ class FunctionsSuite extends MLTest {
         "`org.apache.spark.ml.linalg.Vector` or `org.apache.spark.mllib.linalg.Vector`, " +
         s"but got ${valType}"))
     }
+
+    val df3 = Seq(
+      (Vectors.dense(1.0, 2.0, 3.0), OldVectors.dense(10.0, 20.0, 30.0)),
+      (Vectors.sparse(3, Seq((0, 2.0), (2, 3.0))), OldVectors.sparse(3, Seq((0, 20.0), (2, 30.0))))
+    ).toDF("vec", "oldVec")
+
+    val result3 = df3.select(
+      vector_to_array('vec, dtype = "float32"), vector_to_array('oldVec, dtype = "float32"))
```

Comment:
@WeichenXu123 Do you know a good way to verify the underlying data type is float32?

Reply:
Check array type. Should be "Array<Float>", its corresponding dataframe type is `ArrayType(elementType=FloatType)`

Reply:
or check `sizeof(array[0]) === 4` ?

Reply:
```
scala> df.schema.fields(0).dataType.simpleString
res5: String = array<float>
```
What about checking this dataType.simpleString?